crates/rust-analyzer/src/cli/scip.rs

```diff
@@ -224,6 +265,53 @@ impl flags::Scip {
     }
 }
 
+// TODO: Fix the known buggy cases described here.
+const DUPLICATE_SYMBOLS_MESSAGE: &str = "
+Encountered duplicate scip symbols, indicating an internal rust-analyzer bug. These duplicates are
+included in the output, but this causes information lookup to be ambiguous and so information about
+these symbols presented by downstream tools may be incorrect.
+
+Known cases that can cause this:
+
+  * Definitions in crate example binaries which have the same symbol as definitions in the library
+  or some other example.
+
+  * When a struct/enum/const/static/impl is defined with a function, it erroneously appears to be
+  defined at the same level as the function.
```

Comment:
is this a rust-analyzer bug or a limitation of the SCIP format?

Reply:
I would consider both of these to be rust-analyzer bugs.  The challenge for this one is that the `hir` representation only appears to know what module they are in, and not whether they are contained in a function.  I am not familiar enough with rust-analyzer internals to provide such information.